locale.c

```diff
@@ -5881,48 +5888,68 @@ the program, giving results based on that locale.
 
 =cut
  */
-    PERL_ARGS_ASSERT_MY_STRFTIME;
 
-    /* An empty format yields an empty result */
-    const int fmtlen = strlen(fmt);
-    if (fmtlen == 0) {
-        char *ret;
-        Newxz (ret, 1, char);
-        return ret;
-    }
+STATIC struct tm *
+S_ints_to_tm(pTHX_ int sec, int min, int hour, int mday, int mon, int year,
```

Comment:
Why return a pointer to an allocated `struct tm`?

Either passing in a pointer to a `struct tm` to initialize, or simply returning a `struct tm` itself (not a pointer) would be more efficient and safer -- no need for memory management in the function nor the caller.

Reply:
I understand your first suggestion which I agree is better than this.  But I don't get the second.  Without a pointer to the caller's struct, I don't see how this function could return a full structure.  If it declared it as an automatic, it would go off the stack upon return.  If it declared it as static, then it would be shared memory that could have races.  If it did a Newx, then memory management is involved

Reply:
> But I don't get the second.

You can return structures by value:
```
struct tm mybuildtm(int sec, int min, int hour, int mday, int mon, int year,
                   int wday, int yday, int isdst)
{
    struct tm mytm;
    mytm.tm_sec = sec;
    mytm.tm_min = min;
    mytm.tm_hour = hour;
    mytm.tm_mday = mday;
    mytm.tm_mon = mon;
    mytm.tm_year = year;
    mytm.tm_wday = wday;
    mytm.tm_yday = yday;
    mytm.tm_isdst = isdst;
    mktime(&mytm);
    return mytm;
}
```
is perfectly valid code, though in this case the pass-in-a-pointer seems to [produce better code on gcc](https://godbolt.org/z/h8WbW9z36), clang produces much better code and avoids the copy on return.

Reply:
If I ever knew that, I'd long forgotten.

pushed with the passed pointer version